restservice/src/test/java/org/itevents/service/transactional/MyBatisUserServiceTest.java

```diff
@@ -109,50 +116,53 @@ public void shouldFindAuthorizedUser() {
-        otp.generateOtp(24);
-        String email = "email";
+        otp.generateOtp(OTP_LIFETIME_IN_HOURS);
+        String emailWithOtp = "emailWithOtp";
 
-        when(passwordEncoder.encode(password)).thenReturn(encodedPassword);
-        when(roleService.getRoleByName("subscriber")).thenReturn(BuilderUtil.buildRoleSubscriber());
+        when(roleService.getRoleByName(GUEST_ROLE_NAME)).thenReturn(guestRole);
         doNothing().when(userDao).addUser(eq(testUser), eq(encodedPassword));
-        when(oneTimePassword.generateOtp(24)).thenReturn(otp);
-        when(mailBuilderUtil.buildHtmlFromUserOtp(testUser, otp)).thenReturn(email);
+        when(passwordEncoder.encode(password)).thenReturn(encodedPassword);
+        when(oneTimePassword.generateOtp(OTP_LIFETIME_IN_HOURS)).thenReturn(otp);
+        when(mailBuilderUtil.buildHtmlFromUserOtp(testUser, otp)).thenReturn(emailWithOtp);
 
-        userService.addSubscriber(testUser.getLogin(), password);
+        userService.addSubscriber(testLogin, password);
+
+        verify(roleService).getRoleByName(GUEST_ROLE_NAME);
+        verify(userDao).addUser(testUser, encodedPassword);
+        verify(passwordEncoder).encode(password);
+        verify(oneTimePassword).generateOtp(OTP_LIFETIME_IN_HOURS);
+        verify(userDao).setOtpToUser(testUser, otp);
+        verify(mailService).sendMail(emailWithOtp, testLogin);
 
-        verify(mailService).sendMail(email, testUser.getLogin());
-        verify(userDao).addUser(eq(testUser), eq(encodedPassword));
     }
 
-    /*
-    * TODO: FIX THIS TEST
-    * This test fails for unknown reasons
-    * issue 155
-    * https://github.com/JuniorsJava/itevents/issues/155
-    */
-    @Ignore
-    @Test(expected = EntityAlreadyExistsServiceException.class)
+    @Test
     public void shouldThrowEntityAlreadyExistsServiceExceptionWhenAddExistingSubscriber() throws Exception {
-        User testUser = BuilderUtil.buildUserVlasov();
+        String testLogin = "test@example.com";
+        User testUser = UserBuilder.anUser()
+                .login(testLogin)
+                .role(BuilderUtil.buildRoleGuest())
+                .build();
         String password = "password";
-        String encodedPassword = "encodedPassword";
+        Role guestRole = BuilderUtil.buildRoleGuest();
 
-        when(passwordEncoder.encode(password)).thenReturn(encodedPassword);
-        when(roleService.getRoleByName("subscriber")).thenReturn(BuilderUtil.buildRoleSubscriber());
-        doThrow(new RuntimeException(new SQLIntegrityConstraintViolationException())).when(userDao).addUser(eq(testUser), eq(encodedPassword));
+        when(roleService.getRoleByName(GUEST_ROLE_NAME)).thenReturn(guestRole);
+        doThrow(new RuntimeException(new SQLIntegrityConstraintViolationException()))
+                .when(userDao).addUser(eq(testUser), any(String.class));
+        expectedException.expect(EntityAlreadyExistsServiceException.class);
```

Comment:
@IgorMaksymov see [this comment](https://github.com/JuniorsJava/itevents/commit/2ad8adab018c788454d03e6ca1705498306b4cf3#commitcomment-16675666)